src/lib/commands/create.ts

```diff
@@ -0,0 +1,182 @@
+import {
+  ExtensionContext,
+  commands,
+  FileCreateEvent,
+  WorkspaceEdit,
+  Uri,
+  workspace,
+} from 'vscode';
+import { COMMANDS } from '../constants';
+import { getUniquePathName } from '../fileHelpers';
+const findup = require('findup-sync');
+const { createTemplate } = require('@hubspot/cli-lib/templates');
+const { createModule } = require('@hubspot/cli-lib/modules');
+const { createFunction } = require('@hubspot/cli-lib/functions');
+import * as path from 'path';
+
+// returns a listener that deletes the created file and then allows a given callback to run
+// doAfter receives unique filepath with the given extension
+// ex: extension given is "html", user types "example", but "example.html" already exists. doAfter receives "example1.html"
+const buildFileCreateListener = (
+  extension: string,
+  folderPath: string,
+  doAfter: Function,
+  cleanupCallback: Function
+) => {
+  return async (e: FileCreateEvent) => {
+    if (
+      e.files?.length === 1 &&
+      new RegExp(`${folderPath}/.*`).test(e.files[0].fsPath)
+    ) {
+      cleanupCallback();
+      const uniqueFilePath = getUniquePathName(e.files[0].fsPath, extension);
+      if (e.files[0].fsPath !== uniqueFilePath) {
+        const edit = new WorkspaceEdit();
+        edit.renameFile(e.files[0], Uri.file(uniqueFilePath));
+        await workspace.applyEdit(edit);
+      }
+      doAfter(uniqueFilePath);
+    }
+  };
+};
+
+// doAfter receives (filepath: string)
+const onClickCreateFile = (extension: string, doAfter: Function) => {
+  return (clickContext: any) => {
+    if (clickContext.scheme === 'file') {
+      const createFileSubscription = workspace.onDidCreateFiles(
```

Comment:
There was an issue with module creation using `onDidCreateFiles` instead of `onWillCreateFiles` before. When creating a module folder using "MyModule" in the input, the "MyModule" folder would show up briefly, then get updated to "MyModule.module". Does switching to `onDidCreateFiles` cause a regression?

Reply:
Yep, but this behavior seems unavoidable. `onWillCreateFiles` [doesn't allow edits to the file being created,](https://hubspot.slack.com/archives/C042U2U4NMC/p1671214164062939) and I'm not sure there's another way to do it while still keeping the name prompting in the explorer rather than a popup.

Reply:
Do you mean edits to the file contents? What if we were to have `onClickCreateFolder` use `onWillCreateFiles` and `onClickCreateFile` use `onDidCreateFiles`?

Reply:
We could definitely do that, I figured the additional complexity wasn't worth it for such a minor difference when we wouldn't be able to avoid the behavior for file creation but I'll add it back if you prefer it that way